src/bindings/swift/gen_swift.rs

```diff
@@ -0,0 +1,498 @@
+        // a Rust buffer?
+        let rustBuffer = {{ ci.ffi_bytebuffer_alloc().name() }}(Int64(bytes.count))
+        let pointer = UnsafeMutableBufferPointer(start: rustBuffer.data, count: Int(rustBuffer.len))
+        bytes.copyBytes(to: pointer)
+        self.init(len: Int64(pointer.count), data: pointer.baseAddress!)
+    }
+
+    // Frees the buffer in place. The buffer must not be used after this is
+    // called.
+    func deallocate() {
+        rustbuffer_free(self)
+    }
+}
+
+// A helper class to extract a byte stream from a `Data`.
+
+class DataReader {
+    let data: Data
+    var offset: Data.Index
+
+    init(data: Data) {
+        self.data = data
+        self.offset = 0
+    }
+
+    // Reads a fixed-with integer from the current offset, in big-endian order
+    // to match the Kotlin bindings.
+    func readInt<T: FixedWidthInteger>() -> T {
+        let range = offset..<offset + MemoryLayout<T>.size
+        guard data.count >= range.upperBound else {
+            fatalError()
+        }
+        if T.self == UInt8.self {
+            let value = data[offset]
+            offset += 1
+            return value as! T
+        }
+        var value: T = 0
+        let _ = withUnsafeMutableBytes(of: &value, { data.copyBytes(to: $0, from: range)})
+        offset = range.upperBound
+        return value.bigEndian
+    }
+
+    func hasRemaining() -> Bool {
+        return offset < data.count
+    }
+
+    @inlinable
+    func readUInt8() -> UInt8 {
+        readInt()
+    }
+
+    @inlinable
+    func readUInt32() -> UInt32 {
+        readInt()
+    }
+
+    @inlinable
+    func readUInt64() -> UInt64 {
+        readInt()
+    }
+}
+
+// A helper class to write bytes into a `Data`.
+
+class DataWriter {
+    var bytes: [UInt8]
+    var offset: Array<UInt8>.Index
+
+    init() {
+        self.bytes = []
+        self.offset = 0
+    }
+
+    func writeInt<T: FixedWidthInteger>(_ value: T) {
+        var value = value.bigEndian
+        let _ = withUnsafeBytes(of: &value, { bytes.append(contentsOf: $0) })
+    }
+
+    @inlinable
+    func writeUInt8(_ value: UInt8) {
+        writeInt(value)
+    }
+
+    @inlinable
+    func writeUInt32(_ value: UInt32) {
+        writeInt(value)
+    }
+
+    @inlinable
+    func writeUInt64(_ value: UInt64) {
+        writeInt(value)
+    }
+}
+
+protocol Lowerable {
+    func lower(into buf: DataWriter)
+}
+
+protocol Liftable {
+    static func lift(from buf: DataReader) -> Self
+}
+
+func lowerIntoRustBuffer<T: Lowerable>(_ value: T) -> RustBuffer {
+    let writer = DataWriter()
+    value.lower(into: writer)
+    return RustBuffer(bytes: writer.bytes)
+}
+
+func liftFromRustBuffer<T: Liftable>(_ buf: RustBuffer) -> T {
+    let reader = DataReader(data: Data(rustBuffer: buf))
+    let value = T.lift(from: reader)
+    if reader.hasRemaining() {
+        fatalError()
+    }
+    buf.deallocate()
+    return value
+}
+
+extension Bool : Lowerable, Liftable {
+    func lower() -> UInt8 {
```

Comment:
Add comments explaining the difference between `lower()` and `lower(into:)` (and `lift(_)` and `lift(from:)`).